tests/generation/FileGlobs.py

```diff
@@ -96,3 +96,31 @@ def test_get_project_files_gitignore(self):
 
         os.remove(".gitignore")
         os.chdir(orig_cwd)
+
+
+    def test_get_project_files_ci_mode(self):
+        orig_cwd = os.getcwd()
+        os.chdir(os.path.dirname(os.path.realpath(__file__)))
+        os.makedirs("file_globs_ci_testfiles", exist_ok=True)
+        os.chdir("file_globs_ci_testfiles")
+
+        os.makedirs("src", exist_ok=True)
+        os.makedirs("ignore_dir", exist_ok=True)
+        touch("src", "file.c")
+        touch("root.c")
+        touch("ignore_dir", "src.c")
+        touch("ignore_dir", "src.js")
```

Comment:
why don't you commit all those into git and then you don't need to create them in test?